src/wp-includes/class-wp-theme-json.php

```diff
@@ -2713,58 +2715,77 @@ private static function get_block_nodes( $theme_json, $selectors = array(), $opt
 		}
 
 		foreach ( $theme_json['styles']['blocks'] as $name => $node ) {
-			$selector = null;
-			if ( isset( $selectors[ $name ]['selector'] ) ) {
-				$selector = $selectors[ $name ]['selector'];
-			}
 
-			$duotone_selector = null;
-			if ( isset( $selectors[ $name ]['duotone'] ) ) {
-				$duotone_selector = $selectors[ $name ]['duotone'];
-			}
+			$include_node_paths_only = $options['include_node_paths_only'] ?? false;
+			$node_path = array( 'styles', 'blocks', $name );
+			if ( $include_node_paths_only ) {
+				$nodes[] = $node_path;
```

Comment:
Only returning the array of strings here in _this_ implementation doesn't work well, as it means a different return type based on parameters, which is an anti-pattern. Let's change this to be compatible with the original return type.
```suggestion
				$nodes[] = array(
					'path' => $node_path,
				);
```